tests/bls_onchain_test.go

```diff
@@ -0,0 +1,248 @@
+package tests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/0xsoniclabs/sonic/scc/bls"
+	"github.com/0xsoniclabs/sonic/tests/contracts/blsContracts"
+	gnark "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	"github.com/consensys/gnark-crypto/ecc/bls12-381/fp"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlsVerificationOnChain(t *testing.T) {
+	net := StartIntegrationTestNet(t, IntegrationTestNetOptions{
+		FeatureSet: opera.AllegroFeatures,
+	})
+	defer net.Stop()
+
+	// Deploy contract with transaction options
+	txOptions, err := net.GetTransactOptions(&net.account)
+	require.NoError(t, err, "failed to get transact options; %v", err)
+	txOptions.Nonce = nil
+	txOptions.GasLimit = 10000000
+	blsContract, _, err := DeployContractWithOpts(net, blsContracts.DeployBLS, txOptions)
+	require.NoError(t, err, "failed to deploy contract; %v", err)
+
+	// Test different bls libraries verification
+	tests := []struct {
+		name     string
+		detaFunc func(pk bls.PrivateKey, message []byte) ([]byte, []byte, error)
+	}{
+		{"gnark", getGnarkData},
+		{"bls", getBlsData},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			runTest(t, net, blsContract, test.detaFunc)
```

Comment:
I personally that factoring out test code like here makes them harder to understand. At the call side here, I can not understand what is being tested, and at the definition of the `runTest` function I lack the context of the setup.

Since `runTest` is also not reused, and is a function name in the namespace of this entire package, I would suggest to avoid such helper functions and inline the code here. Wdyt?

Reply:
Simplified test and made it more readable